File: Level-0/46.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int num = 0;
        int f = 1;
        while(true){
            num++;
            f *= num;
            if(f <= n) answer = num;
            if(f > n) break;
        }
        return answer;
    }
}

File: Level-1/46.js
function solution(new_id) {
    const answer = new_id.toLowerCase()
                         .replace(/[^\w-_.]/g, '')
                         .replace(/\.+/g, '.')
                         .replace(/^\.|\.$/g, '')
                         .replace(/^$/g, 'a')
                         .slice(0, 15).replace(/\.$/g, '');
    const len = answer.length;
    return len > 2 ? answer : answer + answer.charAt(len - 1).repeat(3 - len);    
}
